packages/meteor/meteor.d.ts

```diff
@@ -168,26 +203,35 @@ export namespace Meteor {
   >(
     name: string,
     args: ReadonlyArray<EJSONable | EJSONableProperty>,
-    options?: {
-      wait?: boolean | undefined;
-      onResultReceived?:
-        | ((
-            error: global_Error | Meteor.Error | undefined,
-            result?: Result
-          ) => void)
-        | undefined;
-      /**
-       * (Client only) if true, don't send this method again on reload, simply call the callback an error with the error code 'invocation-failed'.
-       */
-      noRetry?: boolean | undefined;
-      returnStubValue?: boolean | undefined;
-      throwStubExceptions?: boolean | undefined;
-    },
+    options?: MethodApplyOptions,
     asyncCallback?: (
       error: global_Error | Meteor.Error | undefined,
       result?: Result
     ) => void
   ): any;
+
+  /**
+   * Invokes a method with an async stub, passing any number of arguments.
+   * @param name Name of method to invoke
+   * @param args Method arguments
+   * @param options Optional execution options
+   * @param asyncCallback Optional callback
+   */
+  function applyAsync<
+    Result extends
+      | EJSONable
+      | EJSONable[]
+      | EJSONableProperty
+      | EJSONableProperty[]
+  >(
+    name: string,
+    args: ReadonlyArray<EJSONable | EJSONableProperty>,
+    options?: MethodApplyOptions,
+    asyncCallback?: (
+      error: global_Error | Meteor.Error | undefined,
+      result?: Result
+    ) => void
```

Comment:
Maybe @denihs is the best person for pinging in this case.
I think applyAsync should not accept a callback.
Regarding the check, that is a great question that we should have discussed.
having the async version with callback can help those projects that just want to migrate and things just work™️ but at the same time blocking the use of it forces the end user to do the migration in the right way I guess

Reply:
We're changing a lot in this function for release 3.0. 

This code wouldn't be valid anymore. I'd wait for the final version before creating the types.

Reply:
If theres going to be any further 2.x releases it would be nice to have some types there, I encountered this when updating a project to 2.12 and wanted to replace some `Meteor.apply()` calls with `Meteor.applyAsync()`.

Reply:
If I understand the api correctly, the promise is used for the stub result, and the callback is used for the server result.

Reply:
Then this should be documented. But honestly I’d expect it to be the opposite? Mostly because the returned `Promise` will be used more than the callback, I think.

Reply:
This is off topic, but for `zodern:relay` I am considering doing something like this so we don't need callbacks:

```
const promise = createProject();
const stubResult = await promise.stubPromise;
const serverResult = await promise;

// Another option.
// The returned object (which also is a promise) is the same thing as the previous example,
// but it also has a property for `serverPromise`
const { stubPromise, serverPromise }  = createProject();
const stubResult = await stubPromise;
const serverResult = await serverPromise;
```

Reply:
Sorry, I missed your message @Grubba27.

The second example still directly returns a promise to get the server result so you still can use `const serverResult = await createProject()`. If the stub result isn't needed, I assume it generally wouldn't be  unnecessarily used since it takes more code. The `stubPromise` and `serverPromise` are properties added to the returned promise.